### PR TITLE
Add use of packrat and packrat introduction

### PR DESCRIPTION
**Pull request by matthewfeickert**
Add instructions on how to use [packrat](https://github.com/rstudio/packrat) for dependency management to ensure a reproducible environment, and add packrat support to the project repo.

### CODE REVIEW

**Comment by matthewfeickert** [comment]
@RaoOfPhysics If you have any feedback on making the short set of instructions more clear, or making the purpose/use of `packrat` more clear that would be very helpful.

**Comment by RaoOfPhysics** [comment]
Ooh/oof. :P

Will take a look, but personally I've avoided `packrat` so far, given I barely understand how to use it automagicaly. I think `binder` uses the `checkpoint` package, which has time-based snapshots of packages from MRAN.

The idea being, if you can do your analysis `$TODAY`, you can set your `checkpoint` snapshot to the date in `YYYY-MM-DD` format, and that'll "guarantee" you can reproduce it in the future.

**Comment by matthewfeickert** [comment]
> Will take a look, but personally I've avoided packrat so far, given I barely understand how to use it automagicaly

Sure. The only reason I'm checking it out is that I've gone down quite the rabbit hole lately when it comes to Python packaging and dependency management with virtual environments. That got me wondering how R does virtual environments and from the searches I did it seems like packrat is the de facto solution.

The most difficult thing for me to understand about packrat is that there is no `update` command like I would expect. `snapshot` _is_ the update command. (insert grumblings about how these API names aren't the API names I would have chosen because I'm a grumpy old man.)

But if you have have time to give any feedback on this, even if it is just on how (un)clear things are that would be great. No rush though. :slightly_smiling_face: 

>  I think binder uses the checkpoint package, which has time-based snapshots of packages from MRAN.

Yup! That's being used in this repo with [`binder/runtime.txt`](https://github.com/matthewfeickert/R-in-Jupyter-with-Binder/blob/master/binder/runtime.txt) providing the snapshot date. :+1:

**Comment by matthewfeickert** [comment]
I'm going to merge this in and if there are edits I can open up a PR to address them in the future. :+1:

**Comment by RaoOfPhysics** [comment]
Hey, sorry, I've had this tab open since you pinged me but haven't had time to review it. Will give it a look as soon as I can, though, and submit any suggestions as a PR. Once again, sorry for not being more prompt.

**Comment by matthewfeickert** [comment]
@RaoOfPhysics You have no need to be sorry! :) I just like your input because you know your stuff and have really good feedback, but I'm in no way wanting to demand time from you (I'm not paying you after all :stuck_out_tongue:). I didn't mean to make you feel rushed by merging it in, I just had forgotten about it myself until I was trolling through my outstanding PR list and thought "eh, might as well merge that in as it isn't 100% awfulness and get an updated Zenodo release that has bug fixes in it." So please don't apologize and please don't feel bad!

File: README.md [hunk 1]
```diff
@@ -40,6 +41,62 @@ Before you can begin you need to make sure that you have the following in your w
...
+
+```
+packrat::clean()
+```
+
+to ensure the minimal environment necessary for reproducibility is kept. Checking the status should now show
+
+```
+packrat::stats()
+Up to date.
```
Comment (RaoOfPhysics) [inline README.md:97]: The output of `R` commands is typically represented by:

```R
#> Up to date.
```

to make it clear that it doesn’t need to be run.

**Review by RaoOfPhysics — COMMENTED** [review]
No major comments, looks good to me.

File: README.md (hunk 1 above)
Comment (matthewfeickert) [reply README.md:97]: This is really helpful to know that is how the community does things. Thanks! I'll open a PR that adds style changes like this.

**Pull request merged**